test/domains/operators/countByBool.chpl

```diff
@@ -0,0 +1,13 @@
+config var flag = true;
+
+var D = {1..10, 1..10};
+var D2 = D#(flag, !flag);
+var D3 = D#(false, true);
+
+writeln(D2);
+writeln(D3);
+
+var A: [D] real;
+
+writeln(A#(flag, !flag));
```

Comment:
Sorry, could you comment why it results in a blank line?  It took me a minute to figure it out (it's because we've cleared one of the dimensions so it's basically empty, right?)

Reply:
That's right.

Reply:
Can you think of a scenario where that's useful?  I'm having trouble and I like to think I'm quite creative with test cases XD

Reply:
I think I would object more strongly to keeping that than the range case

Reply:
I think it's pretty not-very-useful in both cases ("give me either 0 or 1 things from this collection").  I guess it could be considered a simple way to squash something or not (?).  I'm not a big advocate, though, and would not add support for it without a compelling argument from a user if we hadn't already supported it.